presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -828,6 +834,60 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
         metastore.dropTable(session, handle.getSchemaName(), handle.getTableName());
     }
 
+    @Override
+    public ConnectorAnalyzeTableMetadata getAnalyzeTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Object> properties)
+    {
+        HiveTableHandle hiveTableHandle = ((HiveTableHandle) tableHandle).withAnalyzePartitionValues(getPartitionList(properties));
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, hiveTableHandle);
+
+        if (!isCollectColumnStatisticsOnWrite(session)) {
```

Comment:
Remove this. The user shouldn't issue `ANALYZE` if nothing needs to be analyzed.

Reply:
Yea. Plus I think the use may want to disable collect statistics on write but enable column statistics manually by running `ANALYZE`. Will remove the check.